runelite-client/src/main/java/net/runelite/client/plugins/microbot/tempoross/TemporossScript.java

```diff
@@ -0,0 +1,662 @@
+
+        if (TemporossScript.state == State.INITIAL_COOK && doubleFishingSpot != null) {
+            TemporossScript.state = TemporossScript.state.next;
+        }
+
+        if(TemporossScript.ENERGY < 30 && State.getAllFish() > 6 && !temporossConfig.solo() && TemporossScript.state != State.ATTACK_TEMPOROSS) {
+            TemporossScript.state = State.EMERGENCY_FILL;
+        }
+        if (TemporossScript.ENERGY == 0 && !temporossConfig.solo() && TemporossScript.state != State.ATTACK_TEMPOROSS && TemporossScript.state != State.INITIAL_CATCH) {
+            TemporossScript.state = State.ATTACK_TEMPOROSS;
+        }
+        if (temporossPool != null && TemporossScript.state != State.SECOND_FILL && TemporossScript.state != State.ATTACK_TEMPOROSS && TemporossScript.ENERGY < 94) {
+            TemporossScript.state = State.ATTACK_TEMPOROSS;
+        }
+        if (state.isComplete()) {
+            state = state.next;
+            if (state == null) {
+                log("Script looped, resetting to third catch");
+                state = State.THIRD_CATCH;
+            }
+        }
+    }
+
+    private void handleMainLoop() {
+        final Predicate<NPC> filterDangerousNPCs = npc -> !inCloud(npc.getWorldLocation(),1);
+        switch (state) {
+            case INITIAL_CATCH:
+            case SECOND_CATCH:
+            case THIRD_CATCH:
+                isFilling = false;
+                if (inCloud(Microbot.getClient().getLocalPlayer().getWorldLocation(),5)) {
+                    GameObject cloud = sortedClouds.stream()
+                            .findFirst()
+                            .orElse(null);
+                    Rs2Walker.walkNextToInstance(cloud);
+                    Rs2Player.waitForWalking();
+                    return;
+                }
+
+                NPC fishSpot = Rs2Npc.getNpcs(NpcID.FISHING_SPOT_10569)
+                        .filter(npc -> npc.getWorldLocation().distanceTo(workArea.rangePoint) <= 20)
+                        .filter(filterDangerousNPCs)
+                        .findFirst()
+                        .orElse(null);
+
+                if (fishSpot == null) {
+                    fishSpot = Rs2Npc.getNpcs()
+                            .filter(npc -> npc.getId() == NpcID.FISHING_SPOT_10568 || npc.getId() == NpcID.FISHING_SPOT_10565)
+                            .filter(npc -> npc.getWorldLocation().distanceTo(workArea.rangePoint) <= 20)
+                            .filter(filterDangerousNPCs)
+                            .findFirst()
+                            .orElse(null);
+                }
+
+                if (fishSpot != null) {
+                    if (Rs2Player.isInteracting()) {
+                        if (Microbot.getClient().getLocalPlayer().getInteracting().equals(fishSpot)) {
+                            return;
+                        }
+                    }
+                    if(TemporossPlugin.isTethered)
```

Comment:
I see that there are multiple TemporossPlugin.isTethered checks

Not sure if this can be handled at the top of the script once?

Reply:
Holy shit i used an unholy amount of unnecessary isTethered checks, it has been moved